ONSdigital/eq-author-app
# Fix to bug, by adding the `inPlace` parameter to `deepMap`.

### Context of PR ###

When duplicating a section with a `Date Answer` and a `Custom Validation` for either `Earliest Date` or `Latest Date`, the duplicated Page would display an Error Page. 

A module we’re using called `deep-map` is essentially ignoring any value called custom . We’re using it to map through nested objects instead of using some recursion mapping. Luckily you can pass a parameter (`{ inPlace: true }`) to the module telling it not to create a new object but to manipulate the existing object. Which isn’t great practise, but it means it keeps the custom  value. `deep-map` hasn't been updated since 2016 and might not be the best way of altering values in nested objects, but we're only using in one or two places in `eq-author-api`'s `helper.js` file.

### How to review ###

- Create a `Date` answer with a `Custom Validation` within `Advanced Properties`. Can be any number of days, either `earliest date` or `latest date`.
- Duplicate the `section`.
- Check the duplicated `section` and `page` works with no errors.

### What to do after everything is green ###

    *  Bring this branch up-to-date with Origin/Master
    *  If there are a lot of commits or some are not helpful to read, squash them down
    *  Click the Merge button on this pull request
    *  Does this change mean the Capability examples questionnaire should be updated?
    *  Move the Jira ticket for this task into the next stage of the process
